plutus-contract/test/Spec/TimeValidity.hs

```diff
@@ -0,0 +1,119 @@
+
+tests :: TestTree
+tests =
+    testGroup "time validity"
+        [ protocolV5
+        , protocolV6
+        , defaultProtocolParams
+        ]
+
+contract :: Contract () Empty ContractError ()
+contract = do
+    now <- Con.currentTime
+    logInfo @String $ "now: " ++ show now
+    let tx1 = Constraints.mustPayToOtherScript valHash unitDatum $ Ada.lovelaceValueOf 25000000
+    ledgerTx1 <- submitTx tx1
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx1
+    utxos <- utxosAt scrAddress
+    let orefs = fst <$> Map.toList utxos
+        lookups =
+            Constraints.otherScript (validatorScript deadline)
+            <> Constraints.unspentOutputs utxos
+        tx2 =
+            mconcat [Constraints.mustSpendScriptOutput oref unitRedeemer | oref <- orefs]
+            <> Constraints.mustIncludeDatum unitDatum
+            <> Constraints.mustValidateIn (from $ now + 1000)
+    ledgerTx2 <- submitTxConstraintsWith @Void lookups tx2
+    awaitTxConfirmed $ Tx.getCardanoTxId ledgerTx2
+
+trace :: Trace.EmulatorTrace ()
+trace = do
+    void $ Trace.activateContractWallet w1 contract
+    void $ Trace.waitNSlots 3
+
+protocolV5 :: TestTree
+protocolV5 = checkPredicateOptions
+    (defaultCheckOptions & over (emulatorConfig . params . Ledger.protocolParamsL) (\pp -> pp { protocolParamProtocolVersion = (5, 0) }))
+    "tx valid time interval is not supported in protocol v5"
+    (assertFailedTransaction (\_ err _ -> case err of {Ledger.ScriptFailure (EvaluationError ("Invalid range":_) _) -> True; _ -> False  }))
+    (void trace)
+
+protocolV6 :: TestTree
+protocolV6 = checkPredicateOptions
+    (defaultCheckOptions & over (emulatorConfig . params . Ledger.protocolParamsL) (\pp -> pp { protocolParamProtocolVersion = (6, 0) }))
+    "tx valid time interval is supported in protocol v6"
+    (assertValidatedTransactionCount 2)
+    (void trace)
+
+defaultProtocolParams :: TestTree
+defaultProtocolParams = checkPredicateOptions
+    defaultCheckOptions
+    "tx valid time interval is supported in protocol v6+"
+    (assertValidatedTransactionCount 2)
+    (void trace)
+
+deadline :: POSIXTime
+deadline = 1596059092000 -- (milliseconds) transaction's valid range must be after this
+
+{-# INLINEABLE mkValidator #-}
+mkValidator :: P.POSIXTime -> () -> () -> P.ScriptContext -> Bool
+mkValidator dl _ _ ctx = (from dl `contains` range) || P.traceError "Invalid range"
+    where
+    info :: TxInfo
+    info = P.scriptContextTxInfo ctx
+
+    range :: P.POSIXTimeRange
+    range = P.txInfoValidRange info
+
+
+data UnitTest
+instance Scripts.ValidatorTypes UnitTest
+
+typedValidator :: POSIXTime -> Scripts.TypedValidator UnitTest
+typedValidator = Scripts.mkTypedValidatorParam @UnitTest
+    $$(PlutusTx.compile [||mkValidator||])
+    $$(PlutusTx.compile [|| wrap ||])
+    where
+        wrap = Scripts.mkUntypedValidator
+
+validatorScript :: P.POSIXTime -> Validator
+validatorScript = Scripts.validatorScript . typedValidator
```

Comment:
You can simply all of that code to something like:

```
validatorScript :: Plutus.Validator
validatorScript = Plutus.mkValidatorScript
    $$(PlutusTx.compile [|| wrap ||])
 where
     wrap = Scripts.mkUntypedValidator mkValidator

```

Reply:
Unless you plan to use the `TypedValidator` with the constraints library.

Reply:
Am now using `typedValidatorLookups`